vavr/src/main/java/io/vavr/collection/Tree.java

```diff
@@ -143,6 +143,18 @@
     }
 
     /**
+     * Returns a new Node containing the given value and having the children returned from given function.
+     *
+     * @param seed    The start value for the Tree
+     * @param descend A function to calculate the children values
+     * @param <T>     Value type
+     * @return a {@code Tree} with values build up iteration
+     */
+    static <T> Node<T> of(T seed, Function<? super T, Iterable<? extends T>> descend) {
```

Comment:
Please rename the method to `recurse(T seed, Function<? super T, ? extends Iterable<? extends T>> descend)`.

The main reason for renaming it is that `of` might be ambiguous. For example what happens if you change the value type in the unit test from Integer to String? I'm not sure because the call `Tree.of(seed, Stream.....)` matches two signatures: `Tree.of(T, Iterable)` and `Tree(Integer, Function<Integer, ....>)` in the special case of the unit test. This is because Stream is a Seq which implements PartialFunction&lt;Integer, T>, which implements Function<&lt;Integer, T>.

I think the name `Tree.recurse(...)` fits well because  we have `Stream.iterate(...)` which is the same in the sense that a Stream can be seen as Tree with only one or zero children per node.

We use `? extends Iterable` everywhere. I'm not sure if removing `? extends` would still cover all use-cases. (See also our Traversable.flatMap method for example.)